### PR TITLE
Augment the debugging with pry lesson

### PR DESCRIPTION
**Pull request by VictoriaVasys**
@AliSchlereth & @neight-allen if you could take a quick look at this & give me any feedback, I'd appreciate it! I basically just re-ordered a little, added vocab & recap sections and added some more context & additional learning stuff (that I'll touch on, but be brief with). I also built out the example to use an each block since I switched the each section of the lesson to be first.

About how much time do you imagine they should have for the practice section?

### CODE REVIEW

File: module1/lessons/debugging_with_pry.md [hunk 1]
```diff
@@ -1,103 +1,150 @@
...
 
-#### Basic use of .each
+### Basic use of .each
 
 Let's say we have an array of words, and we want to print out to the screen
 each word in the array, but in all capitalized letters.
 
 ```ruby
-array = ["alice", "bob", "eva"]
+names = ["alice", "bob", "eva"]
```
Comment (AliSchlereth) [inline module1/lessons/debugging_with_pry.md]: Love the re-naming!

File: module1/lessons/debugging_with_pry.md [hunk 2]
```diff
@@ -108,9 +155,27 @@ doubles of each number? Triples?
...
+## Recap
+
+* What is the syntax for inline `each`?
+* Multi-line `each`?
+* List 3 debugging techniques for Ruby
+* Where and how do you add 'pry' to your code?
+* What can you do in a pry session?
+
+## Further Learning
+
```
Comment (AliSchlereth) [inline module1/lessons/debugging_with_pry.md]: I like these added resources.

File: module1/lessons/debugging_with_pry.md [hunk 3]
```diff
@@ -108,9 +155,27 @@ doubles of each number? Triples?
...
 * How can you print out the last name and how many characters are in it?
 * How can you create an integer which represents the total number of characters in all the names?
-* Say you had an array of nested arrays: `[['Ilana', 'Corson'], ['Lauren', 'Fazah'], ['Beth', 'Sebian']]`. Each nested array has two elements, a first name and a last name. How would you print out each nested array's full name?
+* Say you had an array of nested arrays: `[['Victoria', 'Vasys'], ['Ali', 'Schlereth'], ['Nate', 'Allen']]`. Each nested array has two elements, a first name and a last name. How would you print out each nested array's full name?
+
+## Recap
+
+* What is the syntax for inline `each`?
+* Multi-line `each`?
+* List 3 debugging techniques for Ruby
```
Comment (AliSchlereth) [inline module1/lessons/debugging_with_pry.md]: Are you looking for byebyg, pry and puts?  I'm only seeing byebug and pry in the lesson plan. Or are you asking what are three ways we can use a debugging tool such as pry?

Comment (VictoriaVasys) [inline module1/lessons/debugging_with_pry.md]: I was thinking pry, puts, & error messages/stack trace bc that's what one of the former debugging lessons highlighted; I think they've seen error messages in their homework (bc the ruby exercises are test-driven, right?), but if not, I'd take byebug as an answer. What do you think?

File: module1/lessons/debugging_with_pry.md (hunk 2 above)
Comment (VictoriaVasys) [inline module1/lessons/debugging_with_pry.md]: 👍

File: module1/lessons/debugging_with_pry.md (hunk 1 above)
Comment (VictoriaVasys) [inline module1/lessons/debugging_with_pry.md]: Thanks! Thought it was a little silly that we *just* taught them about naming conventions & then disobey them here :)

**Comment by AliSchlereth** [comment]
Maybe plan for 15 minutes for practice, they won't get all of it done. I also like the idea of pausing 5min in to practice and either have a student demonstrate how they would use pry in solving one of these or you demo yourself.  

It feels like the intro to each and the intro to debuggers could be 10min each, in which case practice would get more time. Everything moves slower in week 1 than I expect it to though.

**Comment by neight-allen** [comment]
I like what you've got here. It's a lot to get through, but getting the practice time in is important to get it to sink in.

I think I tried to wing this class one morning when Jeff was late. I like to start by putting it in a larger context of "the feedback loop". One of the things that makes us faster developers is shortening it. The shorter the time between asking your code a question, and getting an answer, the quicker you get to working code. Pry is one tool for doing that. Maybe something to kick off with.

**Comment by VictoriaVasys** [comment]
Thanks for the thoughtful comments @neight-allen & @AliSchlereth ! I'll close this PR & create a new one with your suggestions. (In the future, is it best if I create a WIP PR at first?)

**Pull request closed without merging**